### PR TITLE
Fix caching with simple profiler

### PR DESCRIPTION
**Pull request by llin17**
#153

### CODE REVIEW

File: modelkit/core/profilers/simple.py [hunk 1]
```diff
@@ -105,26 +119,61 @@ def summary(  # type: ignore
...
-        result = []
-        data = dict(zip(model_names, model_costs))
-        for model_name in model_names:
-            result.append(
-                max(
-                    data[model_name]
-                    - sum(data.get(sub_model, 0) for sub_model in graph[model_name]),
-                    0,
+        net_duration = duration
+        for sub_model in direct_calls:
```
Comment (tgenin) [inline modelkit/core/profilers/simple.py:174]: NB:
```
        for sub_model, num_calls in direct_calls.items():
            if num_calls > 0:
                net_duration -= sum(self.durations[sub_model][-num_calls:])
```

**Comment by tgenin** [comment]
I think there is an issue in the algorithm: `net_duration -= sum(self.durations[sub_model][-num_calls:])` the lasts calls are not necessarily the calls corresponding to the one to take into account for this computation

**Review by tgenin — CHANGES_REQUESTED** [review]
Thank you, the feature is awesome but I think there is an issue in the algorithm: `net_duration -= sum(self.durations[sub_model][-num_calls:])` the lasts calls are not necessarily the calls corresponding to the one to take into account for this computation

File: modelkit/core/profilers/simple.py (hunk 1 above)
Comment (llin17) [reply modelkit/core/profilers/simple.py]: Thank you for the suggestion, the code has been updated.

**Comment by llin17** [comment]
Thank you for pointing out the issue, the previous commits with `net_duration -= sum(self.durations[sub_model][-num_calls:])` indeed give wrong result in same cases. I have updated the algorithm for the computation of net duration.

**Pull request merged**